### ISSUE TITLE
CSound 6.09.00 breaks repeat section nesting

### ISSUE THREAD

**Issue by Excentertex** [csound/csound#809]
The following score snippet works correctly in CSound 6.07.0rc6, but not in 6.09.00:

```
r 4 J
{ 12 I
i 109 [$I*.2] [.05+.05/11*$I] [500+$I*400] [440-$J*4]	$I	1
}
i 0 [12*.2] .1

r 4 J
{ 12 I
i 109 [$I*.2] [.05+.05/11*$I] [500+$I*400] [440-$J*4]	$I	1
}
i 0 [12*.2] .1
```

The latest version will not recognize $J as a macro.

**Comment by jpffitch** [comment]
I can reproduce.  No idea how t fix yet

**Comment by jpffitch** [comment]
tually J is expanded as can be seen if using -t0 and look at score.srt

i suspect te i 0 statement is causig a early exit, or  it is the anomlous 
end of section -- but note the exta "e"

==John

s
w 0 60
i 109 0.20000000000000001 0.20000000000000001 0.054545499999999997 
0.054545499999999997 900 440 1 1
i 109 0.40000000000000002 0.40000000000000002 0.059090900000000002 
0.059090900000000002 1300 440 2 1
i 109 0.59999999999999998 0.59999999999999998 0.063636399999999996 
0.063636399999999996 1700 440 3 1
i 109 0.80000000000000004 0.80000000000000004 0.068181800000000001 
0.068181800000000001 2100 440 4 1
i 109 1 1 0.072727299999999995 0.072727299999999995 2500 440 5 1
i 109 1.2 1.2 0.0772727 0.0772727 2900 440 6 1
i 109 1.3999999999999999 1.3999999999999999 0.081818199999999994 
0.081818199999999994 3300 440 7 1
i 109 1.6000000000000001 1.6000000000000001 0.086363599999999999 
0.086363599999999999 3700 440 8 1
i 109 1.8 1.8 0.090909100000000007 0.090909100000000007 4100 440 9 1
i 109 2 2 0.095454499999999998 0.095454499999999998 4500 440 10 1
i 109 2.2000000000000002 2.2000000000000002 0.10000000000000001 
0.10000000000000001 4900 440 11 1
i 0 2.3999999999999999 2.3999999999999999 0.10000000000000001 
0.10000000000000001
e
w 0 60
i 109 0 0 0.050000000000000003 0.050000000000000003 500 436 0 1
i 109 0.20000000000000001 0.20000000000000001 0.054545499999999997 
0.054545499999999997 900 436 1 1
i 109 0.40000000000000002 0.40000000000000002 0.059090900000000002 
0.059090900000000002 1300 436 2 1
i 109 0.59999999999999998 0.59999999999999998 0.063636399999999996 
0.063636399999999996 1700 436 3 1
i 109 0.80000000000000004 0.80000000000000004 0.068181800000000001 
0.068181800000000001 2100 436 4 1
.....etc....

**Comment by jpffitch** [comment]
Fixed in git and for 6.09.1
